content/blog/angular-state-management-with-ngxs/index.md

```diff
@@ -0,0 +1,233 @@
+---
+title: "Angular State Management With NGXS"
+date: "2020-09-07"
+coverImage: "title-image.png"
+author: "Akshaya Sharma"
+tags: ["Angular","NGXS","Redux"]
+description: "NGXS is a state management pattern + library for Angular. It acts as a single source of truth for your application's state, providing simple rules for predictable state mutations."
+---
+In this article, we will learn about state management in Angular with NGXS. Here we will understand the basic concept of NGXS and will create a simple TO-DO App.
+
+## What is State
+In the bigger app, to manage the data states is very complicated. In angular, each component has its state, to share the data/state between the components we normally use @Input and @Output decorators, but when the application goes bigger, its challenging to maintain the data consistency. So to solve this problem, redux was introduced. It provides a central store that holds all states of your application. Each component can access the stored state without sending it from one component to another.
```

Comment:
You can change this to `In the bigger apps, managing the data states is very complicated.`